### PR TITLE
fix: Various pessimistic proofs fixes and adaption to `kurtosis-cdk`  pessimistic proof branch

### PR DESCRIPTION
**Pull request by goran-ethernal**
## Description

This PR includes:
1. Fix for replacing `InError` certificates. When a certificate is marked as `InError` on `agglayer`, we need to build a new one with the same height, that includes its bridges and claims, plus new ones that we saw in the meantime.
2. Fix for checking if `Candidate` and `Proven` certificates were also settled on the `agglayer`, because we cannot send a new one, until the previous one is either marked as `InError` or `Settled`.
3. Adapting `local_config` script to the new `kurtosis-cdk` version.
4. Fix where aggsender sent nil for imported bridge exits when it doesnt have any claims.
5. Fix where we didn't rollback `db` transaction in the the `aggsender_storage`.
6. Change db to store hashes as text
7. Add logs 

## Config changes
Parameter `SaveCertificatesToFiles` (bool) have been changed by `SaveCertificatesToFilesPath`: if is = "" is disabled, if not it write certificated in the specific path
```
[aggsender]
SaveCertificatesToFilesPath = ""
```

### CODE REVIEW

File: agglayer/client.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ func (c *AggLayerClient) WaitTxToBeMined(hash common.Hash, ctx context.Context)
 
 // SendCertificate sends a certificate to the AggLayer
 func (c *AggLayerClient) SendCertificate(certificate *SignedCertificate) (common.Hash, error) {
+	if err := defaultNilFieldsInCertificate(certificate); err != nil {
```
Comment (joanestebanr) [inline agglayer/client.go:94]: You need to copy certificate before modified it

Comment (goran-ethernal) [reply agglayer/client.go]: Fix: https://github.com/0xPolygon/cdk/pull/165/commits/e2ff575c631a3a678617fe0172022ad631dae1f3

**Review by joanestebanr — APPROVED** [review]

File: agglayer/types.go [hunk 2]
```diff
@@ -131,19 +153,54 @@ type SignedCertificate struct {
 	Signature *Signature `json:"signature"`
 }
 
+func (s *SignedCertificate) String() string {
+	return fmt.Sprintf("Certificate:%s, \nSignature: %s", s.Certificate.String(), s.Signature.String())
```
Comment (Stefan-Ethernal) [inline agglayer/types.go:157]: Nitpick, in order to align the "labels":

```suggestion
	return fmt.Sprintf("Certificate:%s,\nSignature: %s", s.Certificate.String(), s.Signature.String())
```

File: aggsender/aggsender.go [hunk 3]
```diff
@@ -63,6 +64,8 @@ func New(
 		return nil, err
 	}
 
+	logger.Info(cfg.String())
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender.go:67]: WDYT: add some kind of label/prefix (e.g. `Aggsender config`) to this log?

**Review by Stefan-Ethernal — APPROVED** [review]

File: agglayer/types.go (hunk 2 above)
Comment (goran-ethernal) [reply agglayer/types.go]: Fix: https://github.com/0xPolygon/cdk/pull/165/commits/db602e6fdd89ecedb9aca8526e2b32ef12624a57

File: aggsender/aggsender.go (hunk 3 above)
Comment (goran-ethernal) [reply aggsender/aggsender.go]: Fix: https://github.com/0xPolygon/cdk/pull/165/commits/db602e6fdd89ecedb9aca8526e2b32ef12624a57

File: agglayer/client.go [hunk 4]
```diff
@@ -91,7 +91,9 @@ func (c *AggLayerClient) WaitTxToBeMined(hash common.Hash, ctx context.Context)
 
 // SendCertificate sends a certificate to the AggLayer
 func (c *AggLayerClient) SendCertificate(certificate *SignedCertificate) (common.Hash, error) {
-	response, err := rpc.JSONRPCCall(c.url, "interop_sendCertificate", certificate)
+	certificateToSend := certificate.CopyWithDefaulting()
```
Comment (vcastellm) [inline agglayer/client.go:94]: Is this to the endpoint failing to receive nil values?

**Review by vcastellm — APPROVED** [review]
LGTM
G
T
M

File: agglayer/client.go (hunk 4 above)
Comment (goran-ethernal) [reply agglayer/client.go:94]: Yep. Even though we fixed in the aggsender code that we don't send nil values, for which we wrote a UT `TestSendCertificate_NoClaims`, we left this here as well as a fail safe.

**Pull request merged**